metpy/calc/tools.py

```diff
@@ -28,3 +28,79 @@ def resample_nn_1d(a, centers):
         if index not in ix:
             ix.append(index)
     return ix
+
+def nearest_intersection_idx(x, a, b):
+    """
+    Determines the index of the point just before two lines with
+    common x values.
+
+    Parameters
+    ----------
+    x : array-like
+        1-dimensional array of numeric x-values
+    a : array-like
+        1-dimensional array of y-values for line 1
+    b : array-like
+        1-dimensional array of y-values for line 2
+
+    Returns
+    -------
+        An array of indexes representing the index of the values
+        just before the intersection(s) of the two lines.
+    """
+    # Difference in the two y-value sets
+    difference = a-b
+
+    # Determine the point just before the intersection of the lines
+    # Will return multiple points for multiple intersections
+    sign_change_idx = np.where(np.diff(np.sign(difference)))[0]
+
+    return sign_change_idx
+
+
+def find_intersections(x, a, b):
+    """
+    Calculates the best estimate of the intersection(s) of two y-value
+    data sets that share a common x-value set. Returns list of tuples
+    that are the intersections.
+    """
+
+    # Find the index of the points just before the intersection(s)
+    nearest_idx = nearest_intersection_idx(x, a, b)
+
+    # Build a list of dictionaries of intersections
+    intersections = []
+    for idx in nearest_idx:
+        intersection = {'x1': x[idx],
+                        'x2': x[idx + 1],
+                        'ya1': a[idx],
+                        'ya2': a[idx+1],
+                        'yb1': b[idx],
+                        'yb2': b[idx+1]}
+
+        # Compute the slope of each line
+        intersection['m1'] = (intersection['ya2'] - intersection['ya1'])\
+                             /(intersection['x2']-intersection['x1'])
+
+        intersection['m2'] = (intersection['yb2'] - intersection['yb1'])\
```

Comment:
In general, I much prefer wrapping the expression in an extra set of of `()` rather than using the line continuation character (`\`). The unclosed parenthesis causes the parser automatically to continue onto the next line. (Also when doing so, it's customary to put end the first line with the binary operator, so your mind knows to keep reading onto the next line.)

Of course, once you eliminate the `intersection` dictionary, these should all shorten up quite a bit, probably eliminating all of the multi-line expressions.